src/Type/Schema.php

```diff
@@ -339,15 +341,18 @@ private function loadType(string $typeName) : ?Type
 
         $type = $typeLoader($typeName);
 
-        if (! $type instanceof Type) {
+        if (! $type instanceof Type && ! is_callable($type)) {
```

Comment:
How about we first check if a `callable` is given, call it, then assert that either `$type` or the result of calling the `callable` is a valid `Type`?

Reply:
Sure, if you prefer. 07b9297

Reply:
It's an extra level of safety. Thanks